rust/agama-cli/src/auth.rs

```diff
@@ -37,15 +38,35 @@ pub async fn run(subcommand: AuthCommands) -> anyhow::Result<()> {
 fn read_password() -> Result<String, CliError> {
     let stdin = io::stdin();
     let password = if stdin.is_terminal() {
-        rpassword::prompt_password("Please, introduce the root password: ")?
+        ask_password()?
     } else {
         let mut buffer = String::new();
-        stdin.read_line(&mut buffer)?;
+        stdin
+            .read_line(&mut buffer)
+            .map_err(|_| CliError::MissingPassword)?;
         buffer
     };
     Ok(password)
 }
 
+/// Asks interactively for the password.
+fn ask_password() -> Result<String, CliError> {
+    let validator = |input: &str| {
+        if input.is_empty() {
+            Ok(Validation::Invalid("The password cannot be blank.".into()))
+        } else {
+            Ok(Validation::Valid)
+        }
+    };
+
+    Password::new("Please, introduce the root password:")
+        .with_validator(validator)
+        .without_confirmation()
+        .with_help_message("Press <esc> to exit.")
```

Comment:
I don't see the purpose of this. It helps the user to distinguish entering an empty password from not entering a password?

Reply:
Sorry, I do not get the question

Reply:
The main point is to switch from `rpassword` to `inquire`, which is a more general library for handling user interaction (`rpassword` is only for passwords).

Reply:
I am nitpicking on the existence of this help message. When I remove the help, I can still cancel the operation (and get a CliError::InteractivePassword).

I, as a user, am used to canceling password input by simply entering a garbage or empty password. (Unless the counterparty is a Bank that penalizes all incorrect attempts by $100 and a 3 hour delay until the next attempt, this is a work of fiction :rofl: )

For my personal UX, it is better to hide the help message, because it just draws my attention away towards a useless feature.